web/custom-plugins/forecast-of-response/src/components/Operations/SelectLocation.tsx

```diff
@@ -0,0 +1,148 @@
+  const { location } = props
+  const Map = () => (
+    <Card.Media fullWidth>
+      <img
+        src="https://images.all-free-download.com/images/graphiclarge/europe_map_vectors_design_588720.jpg"
+        alt="map"
+      />
+    </Card.Media>
+  )
+  return (
+    <Div style={{ width: '300px' }}>
+      <Card>
+        <Card.Header>
+          <Card.HeaderTitle>
+            <Heading text="Location on map" variant="h5" />
+          </Card.HeaderTitle>
+        </Card.Header>
+        <Map />
+      </Card>
+    </Div>
+  )
+}
+
+export const SelectOperationLocation = (props: {
+  location: TLocation
+  setLocation: any
+}): JSX.Element => {
+  const [locations, setLocations] = useState<TLocation[]>([])
+  const [
+    searchResult,
+    isLoadingLocations,
+    setSearchResult,
+    hasError,
+  ] = useSearch('ForecastDS/ForecastOfResponse/Blueprints/Location')
+  const [selectLocationType, setSelectLocationType] = useState<string>('select')
+  const { location, setLocation } = props
+
+  /**
+   * Set locations when the search has completed
+   */
+  useEffect(() => {
+    setLocations(searchResult)
+  }, [!isLoadingLocations, searchResult, !hasError])
+
+  return (
+    <>
+      <Div id="opLoc">
+        <Heading text="Enter location" variant="h4" />
+        <Div id="opLoc-type">
+          <LocationButtonsGrid>
+            <Button
+              variant={
+                selectLocationType === 'select' ? 'contained' : 'outlined'
+              }
+              onClick={() => {
+                setSelectLocationType('select')
+              }}
+            >
+              Select existing location
+            </Button>
+            <Button
+              variant={selectLocationType === 'add' ? 'contained' : 'outlined'}
+              onClick={() => {
+                setSelectLocationType('add')
+              }}
+            >
+              Enter UTM coordinates
+            </Button>
+          </LocationButtonsGrid>
+        </Div>
+        <br />
+        {(selectLocationType === 'select' && (
+          <Div id="opLoc-selector">
+            <SingleSelect
+              id="operationLocationSelector"
+              label="Select location"
+              items={locations?.map((loc: TLocation) => {
+                return `${loc.name} - ${loc.UTM}`
+              })}
+              handleSelectedItemChange={(event: any) => {
+                // Parse formatted location string to identify actual loc
+                const locationFormatted = event.selectedItem
+                const [locationName, locationUTM] = locationFormatted.split(
+                  ' - '
+                )
+                const matches = locations?.filter(
+                  (loc: TLocation) =>
+                    loc.name === locationName && loc.UTM === locationUTM
+                )
+                setLocation(matches[0])
+              }}
+            />
+          </Div>
+        )) ||
```

Comment:
A bit strange to have a check on both of these. We should always render something.
I'd rather use `{selectedLocationType==='selected ? <html> : <html>}`